src/Umbraco.Core/StringExtensions.cs

```diff
@@ -541,7 +540,7 @@ public static T EnumParse<T>(this string strType, bool ignoreCase)
         /// <returns>Returns the string without any html tags.</returns>
         public static string StripHtml(this string text)
         {
-            const string pattern = @"<(.|\n)*?>";
+            string pattern = "[*{}\\/:<>?|\"-+()\\n]";

```

Comment:
@Shazwazza the change still needs to be made in order to solve the issue.

Reply:
@elit0451 would you be able to add some unit test cases in the `StringExtensionsTests` to verify that this is stripping Html the way it should? This also prevents us breaking things in the future if we need to change it again.

Reply:
@Shazwazza some unit tests have been added to `StringExtensionsTests` 👍

Reply:
I've reverted the changes to `StripHtml`.

First of all we can't do fundamental changes to a public method without it being a breaking change. Second, regardless of this being a requirement to fix the actual issue we can't just have a `StripHtml` method do something that it isn't supposed to do and that it isn't documented to doing.

The purpose of `StripHtml` is to remove html tags within a string - which it was doing with the old regex. The new regex however simply detects characters that we do not want - and then removes them. It doesn't remove the actual content within the tag (meaning - it will remove `<` and `>` from a `<h1>` tag but then leave the `h1` part intact).

Apart from that - stripping something like `{whatever}` from a string is not what is expected from a `StripHtml` method since this isn't actually a html tag 😃

I've confirmed that the XSS issue is fixed simply by only using the `.CleanForXss()` change that you made. I've also updated the unit test to work with this.